### DISCUSSION TITLE
Is it possible to use gVisor with Talos?

### DISCUSSION THREAD

**Discussion by Sparkxxx** [siderolabs/talos#4984] [Q&A] [answered]
Can I install and use gvisor on Talos, how would this work? 
Thank you

**Comment by frezbo** [comment, marked as answer]
@Sparkxxx please refer this documentation: https://www.talos.dev/docs/v0.15/guides/system-extensions/#example-gvisor

You'll have to be running the `v0.15` alpha release of Talos

**Reply by Sparkxxx** [reply]
Thankx, I want to try Talos and Talos+gVisor as LXC container in Proxmox, do you think it would work, which released cloud image do you recommand?

**Reply by smira** [reply]
you can try `github.com/talos-systems/talos` image, it is used to run Talos in `docker`

see `talosctl cluster create` for an example

please also keep in mind this is not _full_ Talos experience, as container image is still limited

Talos extensions are not supported right now in container mode

**Reply by smira** [reply]
it should be way more flexible to run Talos as a VM in Proxmox, or even on bare metal :)

**Reply by Sparkxxx** [reply]
You are right, I might just waste time trying with lxd for now, maybe come back later, anyway nice project and well thought decisions :)
Thank you

**Reply by Sparkxxx** [reply]
Should gVisor run on ALL including controlplane nodes or only on worker nodes?

**Reply by smira** [reply]
gVisor is a container runtime, it's not something you run on a specific node. You can configure Kubernetes to use gVisor container runtime on per-pod level
